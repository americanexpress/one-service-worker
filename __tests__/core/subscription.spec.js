/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import webPush from 'web-push';

import {
  applicationServerKeyToBase64Array,
  urlBase64ToUint8Array,
} from '../../src/core/subscription';

describe('subscription', () => {
  const VAPIDKeys = webPush.generateVAPIDKeys();
  const applicationServerKey = VAPIDKeys.publicKey;

  describe('applicationServerKeyToBase64Array', () => {
    const vapidPublicKeySample =
      'BEfNAnDvbFoCE1c2ntkuu2_L6KjQr9WbD3KYkq8AIPkdBaFguYx6xgL845CjI8F4PbejwxJ-fHxfrUcLMYlCQZk';

    test('transforms vapid public key to an array of char codes', () => {
      const sample = applicationServerKeyToBase64Array(vapidPublicKeySample);
      // jest/no-large-snapshots...
      expect(JSON.stringify(sample)).toMatchSnapshot();
      expect(sample).toMatchObject(expect.arrayContaining([expect.any(Number)]));
    });
  });

  test('urlBase64ToUint8Array returns Uint8Array', () => {
    expect.assertions(2);

    const typedArray = urlBase64ToUint8Array();
    expect(typedArray).toEqual(expect.any(Uint8Array));
    expect(typedArray.length).toEqual(0);
  });

  test('urlBase64ToUint8Array returns Uint8Array of applicationServerKey', () => {
    expect.assertions(1);

    const typedArray = urlBase64ToUint8Array(applicationServerKey);
    // include padding (=...)
    expect(typedArray.length).toEqual(applicationServerKey.length + 1);
  });
});
