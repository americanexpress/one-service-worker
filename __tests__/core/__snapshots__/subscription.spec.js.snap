// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscription applicationServerKeyToBase64Array transforms vapid public key to an array of char codes 1`] = `"[66,69,102,78,65,110,68,118,98,70,111,67,69,49,99,50,110,116,107,117,117,50,47,76,54,75,106,81,114,57,87,98,68,51,75,89,107,113,56,65,73,80,107,100,66,97,70,103,117,89,120,54,120,103,76,56,52,53,67,106,73,56,70,52,80,98,101,106,119,120,74,43,102,72,120,102,114,85,99,76,77,89,108,67,81,90,107,61]"`;
